Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -50,43 +50,43 @@ def edit
         [HOMEBREW_REPOSITORY]
       end
     else
-      edit_api_message_displayed = T.let(false, T::Boolean)
-      args.named.to_paths.select do |path|
-        core_formula_path = path.fnmatch?("**/homebrew-core/Formula/**.rb", File::FNM_DOTMATCH)
-        core_cask_path = path.fnmatch?("**/homebrew-cask/Casks/**.rb", File::FNM_DOTMATCH)
-        core_formula_tap = path == CoreTap.instance.path
-        core_cask_tap = path == CoreCaskTap.instance.path
+      expanded_paths = args.named.to_paths
+      expanded_paths.each do |path|
+        if path.exist? &&
+            (path.core_formula_path? || path.core_cask_path? || path.core_formula_tap? || path.core_cask_tap?) &&
+            !Homebrew::EnvConfig.no_install_from_api? &&
+            !Homebrew::EnvConfig.no_env_hints?
+          opoo <<~EOS
+              Unless `HOMEBREW_NO_INSTALL_FROM_API` is set when running `brew install`,
+              it will ignore any locally edited #{(path.core_cask_path? || path.core_cask_tap?) ? "casks" : "formulae"}.
+          EOS
+          break
+        end
+      end
 
+      expanded_paths.select do |path|
```

Comment:
I was looking at this a few days ago and I'm not even sure we need a select statement here at all. We never actually select anything in this loop; it's just for validating paths.

Reply:
There is also `end.presence` below that doesn't seem to do anything.

Reply:
Yep, I was wondering about that too.

Reply:
Removed `select`, `precence` and moved the path validation logic into separate `fail` method. Hopefully makes it a little cleaner, and easier to test.